src/Module/Calendar/Event/Form.php

```diff
@@ -0,0 +1,253 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Calendar\Event;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Content\Widget\CalendarExport;
+use Friendica\Core\ACL;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Model\Event as EventModel;
+use Friendica\Model\User;
+use Friendica\Module\Response;
+use Friendica\Module\Security\Login;
+use Friendica\Navigation\SystemMessages;
+use Friendica\Network\HTTPException\BadRequestException;
+use Friendica\Util\ACLFormatter;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Profiler;
+use Friendica\Util\Temporal;
+use Psr\Log\LoggerInterface;
+
+/**
+ * The editor-view of an event
+ */
+class Form extends BaseModule
+{
+	const MODE_NEW  = 'new';
+	const MODE_EDIT = 'edit';
+	const MODE_COPY = 'copy';
+
+	const ALLOWED_MODES = [
+		self::MODE_NEW,
+		self::MODE_EDIT,
+		self::MODE_COPY,
+	];
+
+	/** @var IHandleUserSessions */
+	protected $session;
+	/** @var SystemMessages */
+	protected $sysMessages;
+	/** @var ACLFormatter */
+	protected $aclFormatter;
+	/** @var App\Page */
+	protected $page;
+
+	public function __construct(L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, IHandleUserSessions $session, SystemMessages $sysMessages, ACLFormatter $aclFormatter, App\Page $page, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->session      = $session;
+		$this->sysMessages  = $sysMessages;
+		$this->aclFormatter = $aclFormatter;
+		$this->page         = $page;
+	}
+
+	protected function content(array $request = []): string
+	{
+		if (empty($this->parameters['mode']) || !in_array($this->parameters['mode'], self::ALLOWED_MODES)) {
```

Comment:
What is the case where `$this->parameters['mode']` can be anything than the allowed modes?

Reply:
just a double check in case somebody re-uses this class for other calls :-)